source/extensions/filters/http/admission_control/thread_local_controller.cc

```diff
@@ -17,6 +17,16 @@ ThreadLocalControllerImpl::ThreadLocalControllerImpl(TimeSource& time_source,
                                                      std::chrono::seconds sampling_window)
     : time_source_(time_source), sampling_window_(sampling_window) {}
 
+uint32_t ThreadLocalControllerImpl::averageRps() const {
+  if (historical_data_.empty() || global_data_.requests == 0) {
+    return 0;
+  }
+  using namespace std::chrono;
+  auto count_of_seconds = duration_cast<seconds>(ageOfOldestSample()).count();
+
+  return count_of_seconds == 0 ? 0 : global_data_.requests / count_of_seconds;
```

Comment:
I think you can just do something like below. There's just less going on that way IMO.
```suggestion

using std::chrono::seconds;
seconds secs = std::max(seconds(1), ageOfOldestSample());
return global_data_.requests / secs.count();
```